web_widget_html_markdown/static/src/js/web_widget_html_markdown.js

```diff
@@ -0,0 +1,864 @@
+            }
+
+            if (!assetsLoaded) { // avoid flickering when begin to edit
+                assetsLoaded = new Promise(function (resolve) {
+                    var wysiwyg = new Wysiwyg(self, {});
+                    wysiwyg.attachTo($('<textarea>')).then(function () {
+                        wysiwyg.destroy();
+                        resolve();
+                    });
+                });
+            }
+
+            return Promise.all([this._super(), assetsLoaded, defAsset]);
+        },
+        /**
+         * @override
+         */
+        destroy: function () {
+            delete window.top[this._onUpdateIframeId];
+            if (this.$iframe) {
+                this.$iframe.remove();
+            }
+            this._super();
+        },
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        activate: function (options) {
+            if (this.wysiwyg) {
+                this.wysiwyg.focus();
+                return true;
+            }
+        },
+        /**
+         * Wysiwyg doesn't notify for changes done in code mode. We override
+         * commitChanges to manually switch back to normal mode before committing
+         * changes, so that the widget is aware of the changes done in code mode.
+         *
+         * @override
+         */
+        commitChanges: function () {
+            var self = this;
+            /*
+            if ($(tag).length ==1 ) {
+              self.comment_editing = true;
+              var comment = document.createComment($(tag)[0].outerHTML);
+              $(tag)[0].replaceWith(comment);
+            }
+            */
+
+            if (config.isDebug() && this.mode === 'edit') {
+                var layoutInfo = $.summernote.core.dom.makeLayoutInfo(this.wysiwyg.$editor);
+                $.summernote.pluginEvents.codeview(undefined, undefined, layoutInfo, false);
+            }
+            if (this.mode == "readonly" || !this.isRendered) {
+                return this._super();
+            }
+            var _super = this._super.bind(this);
+            if (self.mode == 'edit' && self.$is_markdown) {
+                // transform in html before saving
+                self.sw_mk_to_html().then(function() {
+                    return self.wysiwyg.saveCroppedImages(self.$content).then(function () {
+                        return self.wysiwyg.save(self.nodeOptions).then(function (result) {
+                            self._isDirty =result.isDirty;
+                            _super();
+                        });
+                    });
+                });
+            } else {
+            // yes repeated block of code, the one above needs to be executed async,
+            // after the wyswyg has rendered
```

Comment:
MMMm okaaaay, but i'm not getting why save needs to be called if mode is not "edit" ? Perhaps this needs to be `else if (self.mode == 'edit')`

Reply:
Putting a pin in this. not closing